packages/@ama-sdk/core/src/plugins/open-telemetry/traceparent/open-telemetry.traceparent.reply.ts

```diff
@@ -0,0 +1,118 @@
+import type {
+  Logger,
+} from '../../../fwk';
+import {
+  PluginRunner,
+  type ReplyPlugin,
+  type ReplyPluginContext,
+} from '../../core';
+import {
+  DEFAULT_COMMON_OPTIONS,
+  type OpenTelemetryTraceparentCommonOption,
+  type OpenTelemetryTransparentReport,
+} from './interfaces';
+
+/** Option for the OpenTelemetry Traceparent Reply plugin */
+export interface OpenTelemetryTraceparentReplyOption extends OpenTelemetryTraceparentCommonOption {
+  /**
+   * Field to be added in the SDK response (default: {@link DEFAULT_OPTIONS.dataField})
+   * @default '_traceparent'
+   */
+  dataField: `_${string}`;
+}
+
+const DEFAULT_OPTIONS = {
+  ...DEFAULT_COMMON_OPTIONS,
+  dataField: '_traceparent'
+} as const satisfies OpenTelemetryTraceparentReplyOption;
+
+/**
+ * Decode and validate a Traceparent Header.
+ * Return undefined if invalid
+ * @param traceparent content of the header
+ * @param options additional options for the function
+ * @param options.logger
+ */
+export const decodeTraceparentHeader = (traceparent: string, options?: { logger?: Logger }) => {
+  const content = traceparent.split('-');
+  const dataField: Partial<OpenTelemetryTransparentReport> = {};
+  const logWarning = () => options?.logger?.warn(`Invalid traceparent header ${traceparent}, will be ignored`);
+
+  if (content.length < 2) {
+    logWarning();
+    return;
+  }
+
+  if (content[0].length < 16) {
+    [dataField.version, dataField.traceId, dataField.parentId, dataField.traceFlags] = content;
+  } else {
+    [dataField.traceId, dataField.parentId, dataField.traceFlags] = content;
+    dataField.version = '00';
+  }
+
+  if (!/^[a-f0-9]{16}$/.test(dataField.parentId) || !/^[a-f0-9]{32}$/.test(dataField.traceId)) {
+    logWarning();
+    return;
+  }
+
+  return dataField as OpenTelemetryTransparentReport;
+};
+
+/**
+ * Plugin to retrieve the {@link https://www.w3.org/TR/trace-context/#traceparent-header | Traceparent header} information for the
+ * {@link https://opentelemetry.io/ | OpenTelemetry} and burn them into the API response object.
+ */
+export class OpenTelemetryTraceparentReply<V = { [key: string]: any } | undefined> implements ReplyPlugin<V, V> {
+  protected readonly options: Readonly<OpenTelemetryTraceparentReplyOption>;
+
+  constructor(options?: Partial<OpenTelemetryTraceparentReplyOption>) {
+    this.options = {
+      ...DEFAULT_OPTIONS,
+      ...options
+    };
+  }
+
+  private updateParentId(parentId?: string, logger?: Logger) {
```

Comment:
I don't think it would make sense to call this method with no parameters
```suggestion
  private updateParentId(parentId: string | undefined, logger?: Logger) {
```